libqtile/extension/dmenu.py

```diff
@@ -19,106 +19,149 @@
+        ("height", None, "defines the height (only supported by some dmenu forks)"),
+    ]
+
+    def __init__(self, **config):
+        base.RunCommand.__init__(self, **config)
+        self.add_defaults(Dmenu.defaults)
+
+    def _configure(self, qtile):
+        base.RunCommand._configure(self, qtile)
+
+        if isinstance(self.command, str):
+            self.configured_command = shlex.split(self.command)
+        else:
+            # Create a clone of self.command, don't use it directly since it's
+            # shared among all the instances of this class
+            self.configured_command = list(self.command)
+
+        if self.bottom:
+            self.configured_command.append("-b")
+        if self.ignorecase:
+            self.configured_command.append("-i")
+        if self.lines:
+            self.configured_command.extend(("-l", str(self.lines)))
+        if self.prompt:
+            self.configured_command.extend(("-p", self.prompt))
+        if self.font:
+            self.configured_command.extend(("-fn", self.font))
+        if self.background:
+            self.configured_command.extend(("-nb", self.background))
+        if self.foreground:
+            self.configured_command.extend(("-nf", self.foreground))
+        if self.selected_background:
+            self.configured_command.extend(("-sb", self.selected_background))
+        if self.selected_foreground:
+            self.configured_command.extend(("-sf", self.selected_foreground))
+        # NOTE: The original dmenu doesn't support the '-h' option
+        if self.height:
+            self.configured_command.extend(("-h", str(self.height)))
+
+    def run(self, items=None):
+        if items:
+            if self.lines:
+                lines = min(len(items), self.lines)
+            else:
+                lines = len(items)
+            self.configured_command.extend(("-l", str(lines)))
+
+        proc = super(Dmenu, self).run()
+
+        if items:
+            input_str = "\n".join([six.u(i) for i in items]) + "\n"
+            return proc.communicate(str.encode(input_str))[0]
+
+        return proc
+
+
+class DmenuRun(Dmenu):
+    """
+    Special case to run applications.
 
     config.py should have something like:
 
     .. code-block:: python
 
         from libqtile import extension
-        mod = 'mod4'
         keys = [
-            ...
-            Key([mod], 'l', lazy.run_extension(extension.WindowList)),
-            Key([mod], 'r', lazy.run_extension(extension.DmenuRun)),
-            ...
+            Key(['mod4'], 'r', lazy.run_extension(extension.DmenuRun(
+                prompt=">",
+                font="Andika-8",
+                background="#15181a",
+                foreground="#00ff00",
+                selected_background="#079822",
+                selected_foreground="#fff",
+                height=24,  # Only supported by some dmenu forks
+            ))),
         ]
```

Comment:
It looks like if you'll have more running extensions with dmenu, then you'll need to duplicate configuration. That's why I coded extension_defaults. To have same look of dmenu everywhere. What do you think about single place of dmenu configuration?